File: lib/models/loan_details.dart
class LoanDetails {
  final bool isEarlyPayoutTaken;
  final bool isLoanProcessed;
  final String vendorUrl;

  LoanDetails({
    required this.isEarlyPayoutTaken,
    required this.isLoanProcessed,
    required this.vendorUrl,
  });

  factory LoanDetails.fromJson(Map<String, dynamic> json) {
    return LoanDetails(
      isEarlyPayoutTaken: json['is_early_payout_taken'] ?? false,
      isLoanProcessed: json['is_loan_processed'] ?? false,
      vendorUrl: json['vendor_url'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'is_early_payout_taken': isEarlyPayoutTaken,
      'is_loan_processed': isLoanProcessed,
      'vendor_url': vendorUrl,
    };
  }
}

File: shared/src/commonMain/kotlin/com/snabbit/runner/shared/features/loan/domain/model/LoanDetails.kt
package com.snabbit.runner.shared.features.loan.domain.model

/**
 * The runner's loan eligibility/status, as the Get-loan bottom sheet needs it
 * (`GET api/v1/runners/me/loan_details`). Pure domain — no serialization.
 *
 * The sheet renders one of three outcomes from these flags (mirrors the Flutter
 * `showLoanUnifiedSheet`):
 *  - [isEarlyPayoutTaken] → "loan not available, you took Early Payout".
 *  - [isLoanProcessed]    → "already processed", CTA opens [vendorUrl].
 *  - otherwise (eligible) → open [vendorUrl] directly (no interstitial).
 */
data class LoanDetails(
    val isEarlyPayoutTaken: Boolean,
    val isLoanProcessed: Boolean,
    val vendorUrl: String,
)
